app/main.py

```diff
@@ -1,2 +1,116 @@
+from typing import Hashable, Any, Iterator
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.__iter = 0
+        self.__capacity = 8
+        self.__threshold = 2 / 3
+        self.__table = [None] * self.__capacity
+        self.__length = 0
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if not isinstance(key, Hashable):
+            raise TypeError(f"unhashable type: {type(key)}")
+
+        index = hash(key) % self.__capacity
+        node = (hash(key), key, value)
+
+        if self.__table[index]:
+            for _ in range(self.__capacity):
+
+                if key == self.__table[index][1]:
+                    self.__table[index] = node
+                    return
+
+                index += 1
+
+                if index == self.__capacity:
+                    index = 0
+
+                if not self.__table[index]:
+                    break
+
+        self.__table[index] = node
+        self.__length += 1
+
+        if self.__length > round(self.__capacity * self.__threshold):
+            self.__double_capacity()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        return self.__find_entry(key, False)
+
+    def __delitem__(self, key: Hashable) -> None:
+        self.__find_entry(key, True)
+
+    def __len__(self) -> int:
+        return self.__length
+
+    def __str__(self) -> str:
+        return str([
+            f"Key: {entry[1]}, Val: {entry[2]}"
+            for entry in self.__table
+            if entry
+        ])
+
+    def __iter__(self) -> Iterator:
+        self.__iter = 0
+        return self
+
+    def __next__(self) -> Any:
+        while self.__iter < self.__capacity and not self.__table[self.__iter]:
```

Comment:
this is not very efficient - it means that every time you call next() you have to iterate through all table cell indexes. Also  `__iter` is not a very good name for a counter variable